src/App.js

```diff
@@ -1,61 +1,87 @@
-  release_date: "2019-12-04",
-  overview: "Some overview that is full of buzzwords to attempt to entice you to watch this movie! Explosions! Drama! True love! Robots! A cute dog!",
-  average_rating: 6,
-  genres: [{id: 18, name: "Drama"}, {id: 42, name: "Comedy"}],
-  budget: 63000000,
-  revenue: 100853753,
-  runtime: 139,
-  tagline: "It's a movie!"
-}
-
 class App extends Component {
+  _isLoaded = false
+
   constructor() {
     super();
+
     this.state = {
-      defaultMovie
+      allMovies: [],
+      currentMovie: this.getMovie(659986), // change this once onClick is fixed, to {}
+      error: ''
     }
   }
 
+  componentDidMount() {
+    this._isLoaded = true;
+
+    fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies`)
+      .then(response => response.json())
+      .then(allMovies => this.setState({ allMovies: allMovies.movies }))
+      .catch(error => this.setState({ error: error.message }))
+  }
+
+  getMovie(id) {
+    this._isLoaded = true; // this isn't working correctly for getMovie, maybe because I'm calling it in the constructor
+
+    fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${id}`)
+      .then(response => response.json())
+      .then(currentMovie => this.setState({ currentMovie: currentMovie.movie }))
+      .catch(error => this.setState({ error: error.message }))
+  }
+
+  // handleClick(id) {
+  //   // this._isLoaded = true; // this isn't working correctly for getMovie, maybe because I'm calling it in the constructor
+  //
+  //   fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${id}`)
+  //     .then(response => response.json())
+  //     // .then(data => console.log(data.movie))
+  //     .then(movieData => this.setState({ currentMovie: movieData.movie }))
+  //     .catch(error => this.setState({ error: error.message }))
+  // }
+
+  showAll() {
+    this.currentMovie = {}
+  }
+
+  componentWillUnmount() {
+    this._isLoaded = false;
+  }
+
+  // todo ==> refactor map over to its own component
   render() {
     return (
       <div className="App">
         <header>
           <img src={logo} className="App-logo" alt="logo" />
           <span className="title">Rancid Tomatillos</span>
-          {/*<section className="cards">*/}
-          {/*  {movieData.movies.map((movie) => {*/}
-          {/*    return <Movie*/}
-          {/*      key={movie.id}*/}
-          {/*      id={movie.id}*/}
-          {/*      poster_path={movie.poster_path}*/}
-          {/*      backdrop_path={movie.backdrop_path}*/}
-          {/*      title={movie.title}*/}
-          {/*      average_rating={movie.average_rating}*/}
-          {/*      release_date={movie.release_date}*/}
-          {/*    />*/}
-          {/*  })}*/}
-          {/*</section>*/}
-          {this.state.defaultMovie &&
-            <section className="profile">
-              <MovieProfile
-                key={defaultMovie.id}
-                data={defaultMovie}
+          <section className="cards">
```

Comment:
You'll also need conditional rendering here - currently, even if your individual movie details are rendering, SO ARE all the movie cards. It's possible you are actually rendering your movie details, but they're just pushed down to the bottom of the page. You'll need some conditions that say "ONLY SHOW all the movies if there's no individual movie to show" or something like that.

Reply:
Yes, I was doing that just so I could see that it worked and to be able to format the movie profile.